test/pthread/test_pthread_proxying_cpp.cpp

```diff
@@ -184,7 +193,10 @@ void test_proxy_async_with_callback(void) {
     queue.proxyAsyncWithCallback(
       returner.native_handle(),
       [&]() {
-        i = 3;
+        {
+          std::unique_lock<std::mutex> lock(mutex);
+          i = 3;
+        }
         executor = std::this_thread::get_id();
         cond.notify_one();
```

Comment:
No, it's actually a pessimization to hold the lock while calling notify because then the other thread wakes up just to immediately block on acquiring the lock.

Reply:
TIL!